js/PixelPainter.js

```diff
@@ -0,0 +1,142 @@
+const pixelPainter = (function() {
+  function gridMaker(width, height, location) {
+    for (let y = 0; y < height; y++) {
+      let row = document.createElement("div");
+      row.className = "row";
+      location.appendChild(row);
+      for (let x = 0; x < width; x++) {
+        let cell = document.createElement("div");
+        cell.className = "cell";
+        cell.setAttribute("data-y", y);
+        cell.setAttribute("data-x", x);
+        row.appendChild(cell);
+      }
+    }
+  }
+
+  function selectColor() {
+    activeColor = this.style.backgroundColor;
+  }
+
+  function inputColor() {
+    activeColor = this.value;
+  }
+
+  function paintClick() {
+    this.style.backgroundColor = activeColor;
+  }
+
+  function paintDrag() {
+    if (mouseDown) {
+      this.style.backgroundColor = activeColor;
+    }
+  }
+
+  function clear() {
+    for (let i = 0; i < gridCells.length; i++) {
+      gridCells[i].style.backgroundColor = "transparent";
+    }
+  }
+
+  function erase() {
+    activeColor = "transparent";
+  }
+
+  return {
+    gridMaker,
+    selectColor,
+    inputColor,
+    paintClick,
+    paintDrag,
+    clear,
+    erase
+  };
+})();
+
+const palette = [
```

Comment:
From here down to your `let mousedown = false` all these variables should be at the top of your closure. Although hoisting will eventually put them there, it's best to manually put them there are well so that people reading your code, know what variables are available to them.

Reply:
Also because these variables aren't within your closure, they are publicly editable by anyone.

Reply:
Pretty much everything below this line should be in your closure.